discovery/spec.md

```diff
@@ -917,108 +931,111 @@ Content-Type: application/json
 
 #### `PUT /services/{id}`
 
-This MUST update the collection of Services in the Discovery Endpoint to
-contain the Service in the request. If the Service identified in the
-request exists it MUST be completely replaced by the Service definition in the
-request. If the Service does not exist then a new Service MUST be created.
+This MUST update, or create, the Service referenced by the specified URL.
+Upon successful processing of the request, any existing Service MUST be
+completely replaced by the Service definition, except where noted below.
 
 The body of the request message MUST contain a Service with an `id`
 attribute matching the `{id}` in the path.
 
+If there is no existing Service with the given `id` then a new Service MUST
+be created with that `id`.
+
 The `epoch` attribute of the Service MAY be omitted. In such cases the
-discovery endpoint MUST assign an appropriate value to the `epoch`. An
-appropriate `epoch` MUST be greater than the any existing `epoch` for a service
-with the same `id`. If an `epoch` is given in the request it MUST be greater
-than the `epoch` of any existing Service of the same `id` or the request MUST
-fail.
-
-The follow responses are defined by this specification:
-- `200 OK` if Service collection was updated.
-  - The HTTP Response Body MUST include the Service values resulting from the
-    successful processing of the request. The response MAY include attributes
-    and values which were added by the Discovery Endpoint.
+discovery endpoint MUST assign a value that is greater than the existing
+`epoch` value for the Service.
+
+If an `epoch` is given in the request and it is not greater than the current
+Service's value, then the request MUST fail.
+
+Aside from `id` and `epoch`, unless there is an out of band agreement,
+all mandatory attributes MUST be present in the request, and a Discovery
+Endpoint MUST reject requests that are missing such attributes.
+
+The Discovery Endpoint MUST ignore attempts to modify read-only or immutable
```

Comment:
Again, there's no way of expressing an intention to modify `id`, so I wouldn't mention that here.

Reply:
ok - removed reference to id